Source/JavaScriptCore/runtime/IteratorOperations.cpp

```diff
@@ -247,6 +247,27 @@ IterationRecord iteratorDirect(JSGlobalObject* globalObject, JSValue object)
     return { object, object.get(globalObject, globalObject->vm().propertyNames->next) };
 }
 
+void validateIterable(JSGlobalObject* globalObject, VM& vm, JSValue iterable, JSValue symbolIterator)
```

Comment:
I don't think `validateIterable` should not throw an error. Should return a something like a enum or something like:

```cpp
enum class IterableValidationResult : uint8_t {
  Valid,
  InvalidString,
  /* ...other types */
}
```

And should throw TypeError in each runtime and LLInt side

What do you think